packages/test/src/ExperimentCrudUtils.ts

```diff
@@ -0,0 +1,171 @@
+}
+
+export async function createSource(server: IntegrationTestServer, sourceName: string, sourceType: string, folderOptions: RequestOptions, userOptions: RequestOptions, auditBehavior: AuditBehaviorTypes = AuditBehaviorTypes.DETAILED, debug?: boolean) : Promise<any[]> {
+    return await insertRows(server, [{ name: sourceName }], 'exp.data', sourceType, folderOptions, userOptions, auditBehavior, debug);
+}
+
+// import, update from file, merge
+export async function importSample(server: IntegrationTestServer, importText: string, sampleType: string, insertOption = "IMPORT", folderOptions: RequestOptions , userOptions: RequestOptions, useAsync = false, debug?: boolean) : Promise<any> {
+    const response = await server.request('experiment', 'importSamples', (agent, url) => {
+            return agent
+                .post(url + '?auditBehavior=DETAILED&crossFolderImport=true')
+                .type('form')
+                .send({
+                    schemaName: 'samples',
+                    queryName: sampleType,
+                    useAsync,
+                    insertOption,
+                    text: importText,
+                    importLookupByAlternateKey: true
+                });
+        },
+        { ...folderOptions, ...userOptions }).expect(successfulResponse);
+
+    if (useAsync)
+        await sleep(100);
+
+    if (debug)
+        console.log(response);
+
+    return response;
+}
+
+// update using api, rowIds will call _update, lsids will call data iterator
+export async function updateRows(server: IntegrationTestServer, rows: any[], schemaName: string, dataType: string, folderOptions: RequestOptions , userOptions: RequestOptions, auditBehavior: AuditBehaviorTypes = AuditBehaviorTypes.DETAILED, debug?: boolean) : Promise<any[]> {
+    const response = await server.post('query', 'updateRows', {
+        schemaName: schemaName,
+        queryName: dataType,
+        rows,
+        auditBehavior,
+    }, { ...folderOptions, ...userOptions }).expect(successfulResponse);
+
+    if (debug)
+        console.log(response);
+
+    return response.body.rows;
+}
+
+// update using api, rowIds will call _update, lsids will call data iterator
+export async function updateSamples(server: IntegrationTestServer, rows: any[], sampleType: string, folderOptions: RequestOptions , userOptions: RequestOptions, auditBehavior: AuditBehaviorTypes = AuditBehaviorTypes.DETAILED, debug?: boolean) : Promise<any[]> {
+    return await updateRows(server, rows, 'samples', sampleType, folderOptions, userOptions, auditBehavior, debug);
+}
+
+export async function saveRows(server: IntegrationTestServer, containerRows: {[key: string] : any[]}, schemaName: string, queryName: string, command: string = 'update', folderOptions: RequestOptions, userOptions: RequestOptions, auditBehavior: AuditBehaviorTypes = AuditBehaviorTypes.DETAILED, debug?: boolean) : Promise<any> {
```

Comment:
many of these helpers seem like more generic functions and aren't experiment specific. Any thoughts on putting them in a general utils file and then having the experiment specific stuff in this file?

Reply:
good point, i've added a todo comment to move non experiment utils